File: GridPatternExample/src/main/scala/org/akka/essentials/grid/controller/RemoteClientEventListener.scala
package org.akka.essentials.grid.controller
import akka.actor.ActorLogging
import akka.actor.Actor
import akka.actor.ActorRef
import akka.remote.RemoteClientError
import akka.remote.RemoteClientConnected
import akka.remote.RemoteClientDisconnected
import akka.remote.RemoteClientStarted
import akka.remote.RemoteClientShutdown
import akka.remote.RemoteClientWriteFailed

class RemoteClientEventListener(val jobScheduler: ActorRef) extends Actor with ActorLogging {

  def receive: Receive = {
    case event: RemoteClientError =>
      log.info("Received remote client error event from address "
        + event.getRemoteAddress());
      jobScheduler ! new StopWorker(event.getRemoteAddress().toString);
      log.info("Cause of the event was {}", event.getCause());
    case event: RemoteClientShutdown =>
      log.info("Received remote client shutdown event from address-> "
        + event.getRemoteAddress().toString);
      jobScheduler ! new StopWorker(event.getRemoteAddress().toString);
    case event: RemoteClientWriteFailed =>
      log.info("Received remote client write fail event from address "
        + event.getRemoteAddress());
      jobScheduler ! new StopWorker(event.getRemoteAddress().toString);
  }
}

File: GridPatternExample/src/main/java/org/akka/essentials/grid/controller/RemoteClientEventListener.java
package org.akka.essentials.grid.controller;

import org.akka.essentials.grid.StopWorker;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import akka.remote.RemoteClientError;

public class RemoteClientEventListener extends UntypedActor {
	private LoggingAdapter log = Logging.getLogger(getContext().system(), this);
	private ActorRef jobScheduler;

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof RemoteClientError) {
			RemoteClientError event = (RemoteClientError) message;
			log.info("Received remote client error event from address "
					+ event.getRemoteAddress());
			jobScheduler.tell(new StopWorker(event.getRemoteAddress()
					.toString()));
			log.info("Cause of the event was {}", event.getCause());
//		} else if (message instanceof RemoteClientWriteFailed) {
//			RemoteClientWriteFailed event = (RemoteClientWriteFailed) message;
//			log.info("Received remote client write fail event from address "
//					+ event.getRemoteAddress());
//			jobScheduler.tell(new StopWorker(event.getRemoteAddress()
//					.toString()));
		}
	}

	public RemoteClientEventListener(ActorRef inJobScheduler) {
		jobScheduler = inJobScheduler;
	}
}
